src/installer/installer.go

```diff
@@ -184,6 +191,42 @@ func (i *installer) InstallNode() error {
 	return i.finalize()
 }
 
+func (i *installer) waitForWorkers(ctx context.Context) error {
+	// This waiting condition is valid only for agent-based installer.
+	kc, err := i.kcBuilder(KubeconfigPath, i.log)
+	if err != nil {
+		i.log.Error(err)
+		return err
+	}
+	if invoker := common.GetInvoker(kc, i.log); invoker != common.InvokerAgent {
+		return nil
+	}
+
+	return utils.WaitForPredicate(waitForeverTimeout, 30*time.Second, func() bool {
+		workers, err := i.inventoryClient.ListsHostsForRole(ctx, string(models.HostRoleWorker))
+		if err != nil {
+			i.log.WithError(err).Error("failed to get workers")
+			return false
+		}
+		if len(workers) == 0 {
+			return true
+		}
+
+		i.log.Infof("Found %d workers, verifying their current installation stage", len(workers))
```

Comment:
From just reading the log output, it is unclear what the wait is for. Perhaps change this log message to indicate that we are waiting for the worker to reach reboot.
```suggestion
		i.log.Infof("Found %d workers, waiting for workers to reach reboot stage", len(workers))
```